packages/ui-contract-editor/src/lib/ContractEditor/index.js

```diff
@@ -78,7 +78,7 @@ const contractProps = {
  */
 const ContractEditor = (props) => {
   const [hoveringFormulaContract, setHoveringFormulaContract] = useState(false);
-  const [formulaDependents, setFormulaDependents] = useState({});
+  const [formulaNode, setFormulaNode] = useState(null);
```

Comment:
This `null` thing confuses me a little bit. Why did it change?

Reply:
This used to be an object that we would set `node` and `dependencies` properties on. Now the node itself has `dependencies` on its data property, so we only need to keep the formula node on state. Initiating it as `null` lets me do the null check on line 90 to ensure we don't try highlighting variables if we don't have a formula node.

Reply:
very nice. thanks for explaining.